### DISCUSSION TITLE
"Patching destroys fonts" - could you give some details?

### DISCUSSION THREAD

**Discussion by rmunn** [kovidgoyal/kitty#8050] [General]
https://sw.kovidgoyal.net/kitty/faq/#kitty-is-not-able-to-use-my-favorite-font mentions that "patching destroys fonts," but doesn't mention any specific details. I'd love to know more: what, specifically, in the original font gets destroyed by the patching process? And what problems does this cause?

Part of why I ask is because I haven't seen this opinion expressed by many other people, so whatever the problems are, they're not widely known. So I feel it would be nice to know more details, so that more people can know that when XYZ happens, it's because of the patched font and here's the solution/workaround.

Any details you could share on the problems with patched fonts would be appreciated.

**Comment by kovidgoyal** [comment]
It depends on how the patching is done. Problems include inconsistent
metrics in monospaced fonts for patched vs unpatched glyphs,
baselines of patched glyphs not lining up, weights of patched characters
not matching, in more sophisticated fonts such as variable fonts you can
even end up with invalid font files. Then things like attachment
positions for accents, leadings, advances, etc all can become
inconsistent leading to poor rendering, overlaps, cut off glyphs and so
on.
